### PR TITLE
Implement copy_entry

### PR DESCRIPTION
**Pull request by dmathieu**

### CODE REVIEW

File: lib/fakefs/fileutils.rb [hunk 1]
```diff
@@ -127,6 +127,10 @@ def copy_file(src, dest, _preserve = false, _dereference = true)
       cp(src, dest)
     end
 
+    def copy_entry(src, dest, _preserve = false, _dereference_root = false, _remove_destination = false)
```
Comment (grosser) [inline lib/fakefs/fileutils.rb]: having all these arguments and then not supporting them seems weird ...

**Comment by grosser** [comment]
why is this needed / how is it different from cp_r ?

**Comment by dmathieu** [comment]
I'm not sure I understand your "why is it needed" question. It's needed because it's in the [FileUtils API](http://ruby-doc.org/stdlib-1.9.3/libdoc/fileutils/rdoc/FileUtils.html#method-c-copy_entry).

The difference with `cp_r` is that this method preserves the file types, as stated in the docs.

**Comment by grosser** [comment]
thats what I meant :)

File: test/fakefs_test.rb [hunk 2]
```diff
@@ -1075,6 +1075,55 @@ def test_can_chmod_R_files
     assert_equal ['/path2'], FileUtils.chmod_R(0600, '/path2')
   end
 
+  def test_copy_entry
```
Comment (dmathieu) [inline test/fakefs_test.rb]: I've simplified this test.

File: lib/fakefs/fileutils.rb (hunk 1 above)
Comment (dmathieu) [inline lib/fakefs/fileutils.rb]: 👍 we're now properly passing them to cp_r.

**Comment by grosser** [comment]
thx! 0.9.3

**Pull request merged**